src/test/java/org/jenkinsci/plugins/workflow/cps/steps/ParallelStepTest.java

```diff
@@ -517,4 +527,178 @@ public void parallelLexicalScope() throws Exception {
+        assertEquals(
+                1,
+                comparator.compare(
+                        new FlowInterruptedException(
+                                Result.FAILURE, new TimeoutStepExecution.ExceededTimeout()),
+                        new AbortException()));
+        assertEquals(-1, comparator.compare(new IllegalStateException(), new AbortException()));
+        assertEquals(0, comparator.compare(new AbortException(), new AbortException()));
+        assertEquals(1, comparator.compare(new AbortException(), new IllegalStateException()));
+        assertEquals(
+                1,
+                comparator.compare(
+                        new FlowInterruptedException(
+                                Result.SUCCESS, new TimeoutStepExecution.ExceededTimeout()),
+                        new FlowInterruptedException(
+                                Result.FAILURE, new TimeoutStepExecution.ExceededTimeout())));
+        assertEquals(
+                -1,
+                comparator.compare(
+                        new FlowInterruptedException(
+                                Result.FAILURE, new TimeoutStepExecution.ExceededTimeout()),
+                        new FlowInterruptedException(
+                                Result.SUCCESS, new TimeoutStepExecution.ExceededTimeout())));
+        assertEquals(
+                1,
+                comparator.compare(
+                        new FlowInterruptedException(
+                                Result.SUCCESS, new TimeoutStepExecution.ExceededTimeout()),
+                        new FlowInterruptedException(
+                                Result.UNSTABLE, new TimeoutStepExecution.ExceededTimeout())));
+        assertEquals(
+                -1,
+                comparator.compare(
+                        new FlowInterruptedException(
+                                Result.UNSTABLE, new TimeoutStepExecution.ExceededTimeout()),
+                        new FlowInterruptedException(
+                                Result.SUCCESS, new TimeoutStepExecution.ExceededTimeout())));
+        assertEquals(
+                1,
+                comparator.compare(
+                        new FlowInterruptedException(
+                                Result.UNSTABLE, new TimeoutStepExecution.ExceededTimeout()),
+                        new FlowInterruptedException(
+                                Result.FAILURE, new TimeoutStepExecution.ExceededTimeout())));
+        assertEquals(
+                -1,
+                comparator.compare(
+                        new FlowInterruptedException(
+                                Result.FAILURE, new TimeoutStepExecution.ExceededTimeout()),
+                        new FlowInterruptedException(
+                                Result.UNSTABLE, new TimeoutStepExecution.ExceededTimeout())));
+    }
+
+    @Issue("JENKINS-49073")
+    @Test
+    public void parallelPropagatesStatus() throws Exception {
```

Comment:
I added these new test cases, which exercise the new functionality and fail prior to this change. Are there any other use cases I should consider or test cases I should write?